p2p/stdnetwork/network.go

```diff
@@ -20,21 +20,22 @@ var (
 type network struct {
 	addrBookMap typesP2P.AddrBookMap
 
-	logger modules.Logger
+	logger *modules.Logger
```

Comment:
The documentation instructs to use the value (not a pointer) to the logger: https://github.com/pokt-network/pocket/wiki/Logger-Readme#module-logging

Since it's a struct (not an interface), I see where you're coming from, but we should probably update the docs & other instances (can be done in a followup commit) if we're going down this approach.

Wdyt?

Reply:
Ok, I took a deeper look and here are my 2c:

## First cent 🪙

Thinking about it, I see little to no value in having an instance of a logger inside a struct. Please correct me if I am missing something. 

this:
```go
type network struct {
	addrBookMap typesP2P.AddrBookMap

	logger modules.Logger
}

func NewNetwork(bus modules.Bus, addrBookProvider providers.AddrBookProvider, currentHeightProvider providers.CurrentHeightProvider) (n typesP2P.Network) {
	networkLogger := logger.Global.CreateLoggerForModule("network")
	networkLogger.Info().Msg("Initializing stdnetwork")

	addrBook, err := addrBookProvider.GetStakedAddrBookAtHeight(currentHeightProvider.CurrentHeight())
	if err != nil {
		networkLogger.Fatal().Err(err).Msg("Error getting addrBook")
	}

	addrBookMap := make(typesP2P.AddrBookMap)
	for _, peer := range addrBook {
		addrBookMap[peer.Address.String()] = peer
	}
	return &network{
		logger:      networkLogger,
		addrBookMap: addrBookMap,
	}
}
```

is probably simpler if it becomes:

```go
var (
	_ typesP2P.Network           = &network{}
	_ modules.IntegratableModule = &network{}
	// 👇👇👇👇👇
	moduleLogger = logger.Global.CreateLoggerForModule("network")
	// 👆👆👆👆👆
)

type network struct {
	addrBookMap typesP2P.AddrBookMap
}

func NewNetwork(bus modules.Bus, addrBookProvider providers.AddrBookProvider, currentHeightProvider providers.CurrentHeightProvider) (n typesP2P.Network) {
	moduleLogger.Info().Msg("Initializing stdnetwork")

	addrBook, err := addrBookProvider.GetStakedAddrBookAtHeight(currentHeightProvider.CurrentHeight())
	if err != nil {
		moduleLogger.Fatal().Err(err).Msg("Error getting addrBook")
	}

	addrBookMap := make(typesP2P.AddrBookMap)
	for _, peer := range addrBook {
		addrBookMap[peer.Address.String()] = peer
	}
	return &network{
		addrBookMap: addrBookMap,
	}
}
```

we would have the `moduleLogger` (or different/better name) defined once inside a package and within that package we would have a zeroLog sublogger that can be used across the board, bypassing any issue in terms of value/reference passing in a "2 birds with one stone way".

Unless we have packages where we have multiple modules, I don't see any drawback with this approach.

As it stands, the `LoggerModule` is not used like the other modules since the module is used to configure something that is subsequently used using a global reference, not something that comes from the bus. 

For simplicity sake, I'd keep it like this (as in keep using logger.Global) but since we are using a global that's public and used across all modules, why not also use a sublogger that's global, **within** the package?

I hope this makes sense.

## Second cent 🪙
Also, I have noticed a `UpdateFields` method used in consensus 

```go
func (m *consensusModule) setLogPrefix(logPrefix string) {
	logger.Global.UpdateFields(map[string]any{
		"kind": logPrefix,
	})
	// INVESTIGATE: Do we need to create a new logger here?
	m.logger = logger.Global.CreateLoggerForModule("consensus")
}
```

It looks wrong, in here I would use the `moduleLogger` approach from above and then instead of calling `setLogPrefix` I would use the built-in zeroLog `With()`: https://github.com/rs/zerolog#sub-loggers-let-you-chain-loggers-with-additional-context


### Suggested plan

1) Merge this now that it doesn't break patterns but fixes an issue that was blocking @bryanchriswhite 
2) Create an issue from this comment
3) Quick discussion if needed
4) Take action / or not depending on 3)

Reply:
> Merge this now that it doesn't break patterns but fixes an issue that was blocking @bryanchriswhite

🆗 

>  I would use the built-in zeroLog With()

👍

> Create an issue from this comment

✅ 

> For consistency I'd apply the change across the board. since it would 100% cause linting issues to anyone that tries to pass the logger inside a function or sub-module if that makes sense.

🧠 

> Take action / or not depending on 3)

You are essentially changing the `moduleLogger` to a `packageLogger` in the context of zerlog's `sublogger`. I see where you're coming from, but it still makes sense to be to have the logger be part of the module.

Example:
1. Things that come from an "instance" of one of our modules, I would want to be namespace (i.e. use the logger).
2. A package might have a "static" function, where we'd be forced to use the global logger rather than the namespace one.
3. If we want to instantiate a module and override it with a noop logger, that'd be possible. 

I don't feel really strongly about this so am okay going with either approach, but am still personally leaning towards the embedded logger approach.